aptos-move/block-executor/src/executor.rs

```diff
@@ -682,13 +712,61 @@ where
                     {
                         data_map.write(key, write_op);
                     }
+
                     // Calculating the accumulated gas costs of the committed txns.
                     let fee_statement = output.fee_statement();
                     accumulated_fee_statement.add_fee_statement(&fee_statement);
                     counters::update_sequential_txn_gas_counters(&fee_statement);
 
+                    // TODO for materialization, we need to understand what we read,
+                    // or do exchange on every transaction, so this logic might change
+                    let mut second_phase = Vec::new();
+                    let mut updates = HashMap::new();
+                    for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                        match change {
+                            AggregatorChange::Create(value) => {
+                                assert!(
+                                    data_map.fetch_aggregator(&id).is_none(),
+                                    "Sequential execution must not create duplicate aggregators"
+                                );
+                                updates.insert(id, value);
+                            },
+                            AggregatorChange::Apply(apply) => {
+                                match apply.get_apply_base_id(&id) {
+                                    ApplyBase::Previous(base_id) => {
+                                        updates.insert(
+                                            id,
+                                            expect_ok(apply.apply_to_base(
+                                                data_map.fetch_aggregator(&base_id).unwrap(),
+                                            ))
+                                            .unwrap(),
+                                        );
+                                    },
+                                    ApplyBase::Current(base_id) => {
+                                        second_phase.push((id, base_id, apply));
+                                    },
+                                };
+                            },
+                        }
+                    }
+                    for (id, base_id, apply) in second_phase.into_iter() {
+                        updates.insert(
+                            id,
+                            expect_ok(apply.apply_to_base(
+                                updates.get(&base_id).cloned().unwrap_or_else(|| {
+                                    data_map.fetch_aggregator(&base_id).unwrap()
+                                }),
+                            ))
+                            .unwrap(),
+                        );
+                    }
+                    for (id, value) in updates.into_iter() {
+                        data_map.write_aggregator(id, value);
```

Comment:
currently this doesn't go anywhere, right? Oh, but it's needed for next txns because you haven't done per-txn ephemeral. I see